File: Java_Design-Patterns/src/facade/PointPolar.java
package facade;

public class PointPolar {
    public double angle;
    private double radius;

    public PointPolar(double r, double a) {
        this.radius = r;
        this.angle = a;
    }

    public void rotate(int angle) {
        this.angle += angle % 360;
    }

    public String toString() {
        return "[" + this.radius + "@" + this.angle + "]";
    }

    public double getAngle() {
        return angle;
    }

    public double getRadius() {
        return radius;
    }
}

File: Python_Design-Patterns/facade/point_polar.py
class PointPolar:

    def __init__(self, r, a):
        self.radius = r
        self.angle = a

    def rotate(self, angle):
        self.angle += angle % 360

    def __str__(self):
        return f'[{self.radius}@{self.angle}]'

    @property
    def angle(self):
        return self.angle

    @property
    def radius(self):
        return self.radius
